pkg/pipelines/tekton_checks.go

```diff
@@ -0,0 +1,113 @@
+package pipelines
+
+import (
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// These CRDs names are checked to confirm that Tekton Pipelines/Triggers has been installed.
+var requiredCRDNames = []string{
+	"pipelines.tekton.dev", "pipelineresources.tekton.dev", "pipelineruns.tekton.dev",
+	"triggerbindings.tekton.dev", "triggertemplates.tekton.dev",
+	"clustertasks.tekton.dev", "taskruns.tekton.dev", "tasks.tekton.dev",
+}
+
+// checkInterface contains a method to return whether Tekton is installed
+type checkInterface interface {
+	checkInstall() (bool, error)
+}
+
+// Strategy to check Tekton install which is checking the existence of CRDs
+type checkStrategy struct {
+	client       v1beta1.CustomResourceDefinitionInterface
+	requiredCRDs []string
+	check        checkInterface
+}
+
+// tektonChecker object that knows how to perform Tekton installation checks
+type tektonChecker struct {
+	strategy *checkStrategy
+}
+
+// newTektonChecker constructs a tektonChecker that is backed by a client configured with user's kubeconfig
+func newTektonChecker() (*tektonChecker, error) {
+	// obtain client config
+	clientConfig, err := getClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// get client set from client config
+	cs, err := clientset.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tektonChecker{
+		strategy: &checkStrategy{
+			requiredCRDs: requiredCRDNames,
+			client:       cs.ApiextensionsV1beta1().CustomResourceDefinitions(),
+		},
+	}, nil
+}
```

Comment:
Maybe the bootstrapper should create the client, and pass it into the TektonChecker?

That way when we come to checking for the `Buildah` cluster task, the pattern repeats?

Reply:
Honestly, it is fine as it.   Checking Buildah requries a different client type anyway.   We will see what type of refactoring is needed as we develop more functionalities.   BTW, bootstrap should not know about what types of tasks are needed.   Checking for tasks will have to be done more locally to where tasks is created.   That would be a different topic.

Reply:
Sounds ok to me.